lesson0.md

```diff
@@ -1,60 +1,106 @@
 # WordPress 0
-** What we will learn today?**
+
+**What we will learn today?**
 - [What is Open Source?](#what-is-open-source)
-- [What is WordPress? WordPress.com versus WordPress.org](#what-is-wordPress-wordPress-com-versus-wordPress-xworg)
+- [What is WordPress? WordPress.com versus WordPress.org](#user-content-what-is-wordpress-wordpresscom-versus-wordpressorg)
 - [What do WordPress developers do?](#what-do-wordpress-developers-do)
 - [The WordPress working environment](#the-wordpress-working-environment) <!-- TODO Choose proper title -->
 - [Homework](#homework)
 ---
 
 ## What is Open Source?
 
-A brief summary of the philosophy behind Open Source. Probably with a link to the video recommended by ? in Slack.
+In 1984, Richard Stallman launched the **GNU Project** and created the **General Public License** (GPL) as a response to his increasing unhappiness with proprietary software, vendor lock-in practices and the resulting loss of personal freedom, as he saw it.
+
+> "When users don't control the program, we call it 'nonfree' or 'proprietary' software. The nonfree program controls the users, and the developer controls the program; this makes the program an instrument of unjust power."
+> - Richard Stallman, [The Free Software Definition](https://www.gnu.org/philosophy/free-sw.html)
+
+This marked the birth of **Free Software**, by which software is considered free if it respects the 4 fundamental freedoms:
+
+1. Anyone has the freedom to run the software as they wish, for any purpose
+2. Anyone has the freedom to study how the program works, and change it as they wish
+3. Anyone has the freedom to redistribute the software
+4. Anyone has the freedom to distribute copies of their own modified version of the Software
+
+As a consequence, access to the code must be open and all 3rd party software needed to run it must also comply with the 4 fundamental freedoms.
+
+A few years later, in 1997, Eric Raymond wrote an essay called **"The Cathedral and the Bazaar"**, in which he described how the Linux operating system and the Free Software philosophy had created a surprisingly effective and collaborative way of writing software.
+
+This set the stage for the appearance, one year later, of the term **Open Source** software. Open Source software is a close relative of Free Software but with a stronger focus on the methodology rather than the philosophy.
```

Comment:
I think it would be good to describe the distinction between copyleft licenses, like GPL, and more flexible open source licenses, like MIT.